docs/madr/decisions/020-kds-redesign.md

```diff
@@ -0,0 +1,242 @@
+* MeshAccessLog
+* MeshCircuitBreaker
+* MeshFaultInjection
+* MeshHealthCheck
+* MeshHttpRoute
+* MeshProxyPatch
+* MeshRateLimit
+* MeshRetry
+* MeshTimeout
+* MeshTrace
+* MeshTrafficPermission
+
+When there are two policies, one at the zone level and another at the global level, that impact the same dataplanes, the more specific policy takes precedence. In this scenario, the Zone policy would be the one utilized.
+
+Global policy:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+ name: timeout-global
+ namespace: kuma-system
+ labels:
+   kuma.io/mesh: default
+spec:
+ targetRef:
+   kind: Mesh
+ to:
+   - targetRef:
+       kind: Mesh
+     default:
+       idleTimeout: 20s
+       connectionTimeout: 2s
+```
+
+Zone policy:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+ name: timeout-zone
+ namespace: kuma-system
+ labels:
+   kuma.io/mesh: default
+spec:
+ targetRef:
+   kind: Mesh
+ to:
+   - targetRef:
+       kind: Mesh
+     default:
+       idleTimeout: 10s
+       connectionTimeout: 2s
+```
+
+The policy `timeout-zone` is going to be applied to all dataplanes in the default mesh in the zone. Also, worth adding that global control-plane added `kuma.io/zone: zone-name` to the resource when is synced.
+
+### Incremental xDS
+
+Incremental xDS in the `DeltaDiscoveryRequest` has a field `resource_names_subscribe` which requires information about resources that we want to subscribe to e.g. policy with name `timeout-1`. If you send `resource_names_subscribe:  "*"` in a `DeltaDiscoveryRequest`, the control plane should respond with all resources of the specified type, not just the ones that have changed since the last update. This is because the `resource_names_subscribe` field is used to filter which resources the control plane should send, and if it is set to `*`, it effectively disables filtering and requests all resources of the specified type. Looks like go-control-plane has no implementation of the client, so it might requires us to implement one.
+
+xDS Specification
+* https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#how-the-client-specifies-what-resources-to-return
+* https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#incremental-xds
+* https://github.com/envoyproxy/go-control-plane/blob/main/pkg/cache/v3/delta.go#L31
+
+### Measure performance of current KDS solution
+
+As a part of a task we could consider mesuring performance of the KDS to compare it with the new solution.
+We should take a look at:
+* control-planes CPU and Memory usage
+* Latency
+* bandwidth usage 
+
+## How we can split the work
+
+We can split the implementation and introduce features independently.
+
+### Stages
+
+1. Introduce `ZoneToGlobalSync` with incremental xDS
+2. Introduce `GlobalToZoneSync` with incremental xDS
+3. Prioritize resource synchronization
+4. Introduce handshake (if required)
+5. Enable the creation of resources on the zone - separate MADR
```

Comment:
As a part of KDS redesign, I'd also consider this https://github.com/kumahq/kuma/issues/3247